ocrd_network/ocrd_network/processor_server.py

```diff
@@ -0,0 +1,205 @@
+            tags=['Processing'],
+            status_code=status.HTTP_200_OK,
+            summary='Get information about this processor.',
+            response_model=PYOcrdTool,
+            response_model_exclude_unset=True,
+            response_model_exclude_none=True
+        )
+
+        self.router.add_api_route(
+            path='/',
+            endpoint=self.create_processor_job_task,
+            methods=['POST'],
+            tags=['Processing'],
+            status_code=status.HTTP_202_ACCEPTED,
+            summary='Submit a job to this processor.',
+            response_model=PYJobOutput,
+            response_model_exclude_unset=True,
+            response_model_exclude_none=True
+        )
+
+        self.router.add_api_route(
+            path='/{job_id}',
+            endpoint=self.get_processor_job,
+            methods=['GET'],
+            tags=['Processing'],
+            status_code=status.HTTP_200_OK,
+            summary='Get information about a job based on its ID',
+            response_model=PYJobOutput,
+            response_model_exclude_unset=True,
+            response_model_exclude_none=True
+        )
+
+    async def startup(self):
+        await initiate_database(db_url=self.db_url)
+        DBProcessorJob.Settings.name = self.processor_name
+
+    async def get_processor_info(self):
+        if not self.ocrd_tool:
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f'Empty or missing ocrd_tool'
+            )
+        return self.ocrd_tool
+
+    # Note: The Processing server pushes to a queue, while
+    #  the Processor Server creates (pushes to) a background task
+    async def create_processor_job_task(self, job_input: PYJobInput, background_tasks: BackgroundTasks):
+        validate_job_input(self.log, self.processor_name, self.ocrd_tool, job_input)
+        job_input = await validate_and_resolve_mets_path(self.log, job_input, resolve=True)
+
+        job_id = generate_id()
+        job = DBProcessorJob(
+            **job_input.dict(exclude_unset=True, exclude_none=True),
+            job_id=job_id,
+            processor_name=self.processor_name,
+            state=StateEnum.queued
+        )
+        await job.insert()
+
+        execution_failed = False
+        start_time = datetime.now()
+        await db_update_processing_job(
+            job_id=job_id,
+            state=StateEnum.running,
+            start_time=start_time
+        )
+        try:
+            background_tasks.add_task(
+                run_single_execution,
+                ProcessorClass=self.ProcessorClass,
+                executable=self.processor_name,
+                abs_path_to_mets=job.path_to_mets,
+                input_file_grps=job.input_file_grps,
+                output_file_grps=job.output_file_grps,
+                page_id=job.page_id,
+                parameters=job.parameters
+            )
+        except Exception as error:
```

Comment:
https://github.com/OCR-D/core/blob/93482d39e50ded9046cae09e0bc48969ab8d28e4/ocrd_network/ocrd_network/processor_server.py#L143-L150

Yes, external queues are not an option. We definitely need to run within the same process here, otherwise instance caching would not work. @tdoan2010's design choice was to not allow concurrency in these servers (multiple workers via multithreading or multiprocessing backed by shared memory, as in my [Processing Server](https://github.com/OCR-D/core/pull/652)), but instead require running multiple instances if necessary.

But to address FastAPI's problem with error handling (follow-up tasks getting lost) and oversubscription (no requests getting denied), couldn't we also just make the Processor Server a simple blocking API?

Reply:
> couldn't we also just make the Processor Server a simple blocking API

Sure, that will simplify a lot of things. But we need to make sure that the `timeout` is set properly then. This can be a parameter set by users when they start up a server.

Reply:
@MehmedGIT you now [implemented](https://github.com/OCR-D/core/pull/1030/commits/4b5d48ea680238059e4e71d39a63ed63f364d92e) this as a client-side timeout (with a dummy for the number of pages). I could live with that, but we must not forget to
- catch the additional ReadTimeout
- make this configurable (in the Processing Server now; and in the Processor Server client CLI)
- use the actual number of pages (by resolving the job's `page_id` param against its `workspace_id` or `path_to_mets` param, and then counting the remaining number of physical pages on the workspace)

In contrast, if we do place this on the side of the Processor Server as @tdoan2010 proposed (I guess that would be around the [`invoke_processor` call](https://github.com/OCR-D/core/blob/4b5d48ea680238059e4e71d39a63ed63f364d92e/ocrd_network/ocrd_network/processor_server.py#L143)), we would have the chance of passing the timeout param down (over API or CLI) to the actual Processor instance. (Which the Processing Worker could also benefit from BTW.) So in that scenario, on the caller side, we would get an additional error case (timeout).

Reply:
> I thought dummy timeout for now is still better than no timeout at all.

Yes, it's good to have it in the code already. Perhaps I should have created an issue right away instead of prolonging the discussion here.

> > catch the additional ReadTimeout
> 
> Could you elaborate on that a bit more? 

If you set a [`timeout`](https://www.python-httpx.org/advanced/#timeout-configuration) on httpx.Client or AsyncClient, then it can raise TimeoutException (or perhaps one of its [subclasses](https://www.python-httpx.org/exceptions/), which I assume would have to be ReadTimeout here, because the connection itself already has been established and we are running out of time while trying to read the response).

> I am just not sure what is best for now. Providing an extra field for request timeout or for amount of pages, so the timeout could be then calculated.

I am convinced generic timeouts make no sense for us, so calculation relative to number of pages is crucial.

> When there is a page range, sure. What about the cases when no `page_id` is provided?

Even if there is a page range parameter value (list or regexp), it must be resolved against the actual pages, which yields the true list (that could be shorter than the input list or empty). So regardless, one needs access to the workspace instance (deeper inside the call tree). Perhaps the parameter validator could also give us that number already.

Reply:
OK, for now we have the hardcoded 200 pages and a TODO, let's not let that hinder merging now and revisiting for proper page list information from the workspace later.